src/audio/kpb.c

```diff
@@ -453,6 +457,10 @@ static struct comp_dev *kpb_new(const struct comp_driver *drv,
 	kpb->hd.c_hb = NULL;
 	kpb->kpb_no_of_clients = 0;
 	kpb->state_log = 0;
+	/* mask is set to zero to allow legacy coping for default config */
+	kpb->mic_sel_->mask_ = 0x0;
+	kpb->mic_sel_cnt_ = 0;
+	kpb->num_of_in_channels_ = 0;
```

Comment:
in fact all these 0 assignments are redundant - `kpb` is allocated using `rzalloc()`. Using `rzalloc()` means we *want* it to be 0 and we rely on it and we use it. Not just do it "in case we forget something." I'd rather remove the above assignments instead of adding new ones.